src/lib.rs

```diff
@@ -751,6 +751,76 @@ impl Compressor {
     pub fn symbol_lengths(&self) -> &[u8] {
         &self.lengths[0..self.n_symbols as usize]
     }
+
+    /// Rebuild a compressor from an existing symbol table.
+    ///
+    /// This will not attempt to optimize or re-order the codes.
+    pub fn rebuild_from(symbols: impl AsRef<[Symbol]>, symbol_lens: impl AsRef<[u8]>) -> Self {
+        let symbols = symbols.as_ref();
+        let symbol_lens = symbol_lens.as_ref();
+
+        assert_eq!(
+            symbols.len(),
+            symbol_lens.len(),
+            "symbols and lengths differ"
+        );
+        assert!(symbols.len() <= 254, "symbol table len must be <= 254");
+        validate_symbol_order(symbol_lens);
+
+        // Insert the symbols in their given order into the FSST lookup structures.
+        let symbols = symbols.to_vec();
```

Comment:
Why do we need to copy all the symbols here since we just iterate them later?

Reply:
we have to own them, they are part of the returned compressor